frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -0,0 +1,708 @@
+	}
+
+	/// Helper function for the case where a solution is accepted in the signed phase.
+	///
+	/// Extracted to facilitate with weight calculation.
+	///
+	/// Infallible
+	pub fn finalize_signed_phase_accept_solution(
+		ready_solution: ReadySolution<T::AccountId>,
+		who: &T::AccountId,
+		deposit: BalanceOf<T>,
+		reward: BalanceOf<T>,
+	) {
+		// write this ready solution.
+		<QueuedSolution<T>>::put(ready_solution);
+
+		// unreserve deposit.
+		let _remaining = T::Currency::unreserve(who, deposit);
+		debug_assert!(_remaining.is_zero());
+
+		// Reward.
+		let positive_imbalance = T::Currency::deposit_creating(who, reward);
+		T::RewardHandler::on_unbalanced(positive_imbalance);
+	}
+
+	/// Helper function for the case where a solution is accepted in the rejected phase.
+	///
+	/// Extracted to facilitate with weight calculation.
+	///
+	/// Infallible
+	pub fn finalize_signed_phase_reject_solution(who: &T::AccountId, deposit: BalanceOf<T>) {
+		let (negative_imbalance, _remaining) = T::Currency::slash_reserved(who, deposit);
+		debug_assert!(_remaining.is_zero());
+		T::SlashHandler::on_unbalanced(negative_imbalance);
+	}
+
+	/// Insert a solution into the queue while maintaining an ordering by solution quality.
+	///
+	/// Solutions are ordered in reverse: strong solutions have low indices.
+	///
+	/// If insertion was successful, `Some(index)` is returned where index is the
+	/// index of the newly inserted item.
+	///
+	/// Note: this function maintains the invariant that `queue.len() <= T::SignedMaxSubmissions`.
+	/// In the event that insertion would violate that invariant, the weakest element is dropped.
+	///
+	/// Invariant: The returned index is always a valid index in `queue` and can safely be used to
+	/// inspect the newly inserted element.
+	pub fn insert_submission(
+		who: &T::AccountId,
+		queue: &mut Vec<SignedSubmission<T::AccountId, BalanceOf<T>, CompactOf<T>>>,
+		solution: RawSolution<CompactOf<T>>,
+		size: SolutionOrSnapshotSize,
+	) -> Option<usize> {
+		// Let's ensure that our input is valid.
+		let max_submissions = T::SignedMaxSubmissions::get();
+		debug_assert!(queue.len() as u32 <= max_submissions);
+
+		let threshold = T::SolutionImprovementThreshold::get();
+		// this insertion logic is a bit unusual in that a new solution which beats an existing
+		// solution by less than the threshold is sorted as "greater" than the existing solution.
+		// this means that the produced ordering depends on the order of insertion, and that
+		// attempts to produce a total ordering using this comparitor are highly unstable.
+		//
+		// this ordering prioritizes earlier solutions over slightly better later ones.
+		let insertion_position = queue.binary_search_by(|s| {
+			if is_score_better::<Perbill>(
+				solution.score,
+				s.solution.score,
+				threshold,
+			) {
+				Ordering::Greater
+			} else {
+				Ordering::Less
+			}
+		}).expect_err("comparitor function never returns Ordering::Equal; qed");
```

Comment:
is score better is not `comparitor` per se, but you can also check its implementation and make the same claim.